packages/core/content-type-builder/admin/src/components/FormModalSubHeader/index.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { Typography } from '@strapi/design-system/Typography';
+import upperFirst from 'lodash/upperFirst';
+import { getTrad } from '../../utils';
+import getModalTitleSubHeader from './getModalTitleSubHeader';
+
+const FormModalSubHeader = ({
+  actionType,
+  modalType,
+  forTarget,
+  kind,
+  step,
+  attributeType,
+  attributeName,
+  customField,
+}) => {
+  const { formatMessage } = useIntl();
+  const type =
```

Comment:
I think this should be called `isCustomField` or something like that.

Reply:
I think it should remain `type` since it doesn't return a boolean and the object shorthand can be used in the formatMessage variables object. Maybe you meant just below for `modalType === 'customField'`?

Reply:
Or maybe something like this 🤷‍♂️ ?
```javascript
const isCustomField = modalType === 'customField';
const intlLabel = isCustomField
  ? customField.intlLabel
  : { id: getTrad(`attribute.${attributeType}`)};
//...
{
  type: upperFirst(formatMessage(intlLabel)),
  name: upperFirst(attributeName),
  step,
}
```

Reply:
I don't see the original code any longer, but when I reviewed it I had the impression that `type` is a boolean holding the information whether it is a custom field or not. If that is not the case, feel free to resolve this discussion :)

You could also shorten this to:

```js
{
  type: upperFirst(formatMessage(customField?.intlLabel ?? { id: getTrad(`attribute.${attributeType}`)})),
  name: upperFirst(attributeName),
  step,
}
```